src/attr/_make.py

```diff
@@ -156,6 +162,14 @@ def attrib(default=NOTHING, validator=None,
         )
         converter = convert
 
+    if factory is not None:
+        if default is not NOTHING:
+            raise ValueError(
+                "The `default` and `factory` arguments are mutually "
+                "exclusive."
+            )
+        default = Factory(factory)
```

Comment:
for sanity, can factory fail if its being passed something other than a callable?
i can already see people writing `factory=Factory(...)`